File: backend/app/modeles/application.py
from datetime import datetime
from app.db import db

class Application(db.Model):
    __tablename__ = 'applications'

    id = db.Column(db.Integer, primary_key=True)
    title = db.Column(db.String(150), nullable=False)
    company = db.Column(db.String(150), nullable=False)
    status = db.Column(db.Enum('Wishlist', 'Applied', 'Interview', 'Offer', 'Rejected', name='status_enum'), nullable=False)
    date_applied = db.Column(db.Date, nullable=True)
    job_board = db.Column(db.String(100), nullable=True)
    location = db.Column(db.String(100), nullable=True)
    notes = db.Column(db.Text, nullable=True)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)

    def to_dict(self):
        return {
            "id": self.id,
            "title": self.title,
            "company": self.company,
            "status": self.status,
            "date_applied": self.date_applied.isoformat() if self.date_applied else None,
            "job_board": self.job_board,
            "location": self.location,
            "notes": self.notes,
            "created_at": self.created_at.isoformat()
        }

File: frontend/src/types/application.ts
// src/types/application.ts

export type ApplicationStatus = 'Wishlist' | 'Applied' | 'Interview' | 'Offer' | 'Rejected';

export interface Application {
  id: number;
  title: string;
  company: string;
  status: ApplicationStatus;
  date_applied: string | null;
  job_board?: string | null;
  location?: string | null;
  notes?: string | null;
  created_at: string;
  logo?: string;
}


export interface GroupedApplications {
  Wishlist: Application[];
  Applied: Application[];
  Interview: Application[];
  Offer: Application[];
  Rejected: Application[];
}
